src/js/merkle-drop/ClaimFlow.js

```diff
@@ -99,40 +106,54 @@ function ClaimFlow() {
       })
   }, [])
 
-  const claim = useCallback(() => {
-    closeTermsAndConditionsModal()
-    setInternalState(STATE.CLAIM_START)
-    requestPermission()
-      .then(permission => {
-        if (!permission) {
+  const claim = useCallback(async () => {
+    try {
```

Comment:
If you rewrite something like this with try catch, you have to make sure it works. 
(Spoiler, it does not work)

Reply:
Your catch block does not catch the exceptions anymore